R/format_data.R

```diff
@@ -2414,6 +2414,676 @@ fmt_datetime <- function(data,
   )
 }
 
+#' Format values as time durations
+#'
+#' @description
+#' Format input values to time durations whether those input values are numbers
+#' or of the `difftime` class. We can specify which time units any numeric input
+#' values have (as days, hours, minutes, or seconds) and the output can be
+#' customized with a duration style (corresponding to short, long, and
+#' colon-separated forms) and a choice of output units from weeks to seconds.
+#'
+#' @details
+#' Targeting of values is done through `columns` and additionally by `rows` (if
+#' nothing is provided for `rows` then entire columns are selected). Conditional
+#' formatting is possible by providing a conditional expression to the `rows`
+#' argument. See the *Arguments* section for more information on this.
+#'
+#' @inheritParams fmt_number
+#' @param input_units If one or more selected columns contains numeric values, a
+#'   keyword must be provided for `input_units` for **gt** to determine how
+#'   those values are to be interpreted in terms of duration. The accepted units
+#'   are: `"seconds"`, `"minutes"`, `"hours"`, `"days"`, and `"weeks"`.
+#' @param output_units Controls the output time units. The default, `NULL`,
+#'   means that **gt** will automatically choose time units based on the input
+#'   duration value. To control which time units are to be considered for output
+#'   (before trimming with `trim_zero_units`) we can specify a vector of one or
+#'   more of the following keywords: `"weeks"`, `"days"`, `"hours"`,
+#'   `"minutes"`, or `"seconds"`.
+#' @param duration_style A choice of four formatting styles for the output
+#'   duration values. With `"narrow"` (the default style), duration values will
+#'   be formatted with single letter time-part units (e.g., 1.35 days will be
+#'   styled as `"1d 8h 24m`). With `"wide"`, this example value will be expanded
+#'   to `"1 day 8 hours 24 minutes"` after formatting. The `"colon-sep"` style
+#'   will put days, hours, minutes, and seconds in the `"([D]/)[HH]:[MM]:[SS]"`
+#'   format. The `"iso"` style will produce a value that conforms to the ISO
+#'   8601 rules for duration values (e.g., 1.35 days will become `"P1DT8H24M"`).
```

Comment:
For `colon-sep` it would be great to respect `output_units`. This would have to be slightly different than narrow/wide: `c("hours", "seconds")` would not be valid, for example (or probably any skipping of units, period). For `trim_zero_units`, the only valid values should be `FALSE` and `"leading"`, and `trim_zero_units` is ignored.